### PR TITLE
IDVA6-1629 implement csrf tokens

### PR DESCRIPTION
**Pull request by mmodzelewski-ch**
Link to Jira:
https://companieshouse.atlassian.net/browse/IDVA6-1629
https://companieshouse.atlassian.net/browse/IDVA6-1748

Changes:
- CSRF tokens implemented
- hidden input fields added to forms
- unit tests fixed and new added
- dependencies updated

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by bwallace-ch — DISMISSED** [review]

File: src/app.ts [hunk 1]
```diff
@@ -90,13 +101,13 @@ app.use((req: Request, res: Response, next: NextFunction) => {
 routerDispatch(app);
 
 // http-error error handler
-app.use(httpErrorHandler);
+app.use(...Object.values(errorHandler));
```
Comment (tferns-ch) [inline src/app.ts:104]: Not a fan of using `...Object.values` here, would prefer an explicit import and reference to the handlers i.e:

```suggestion
app.use(errorHandler.httpErrorHandler);
app.use(errorHandler.csrfErrorHandler);
```

Just makes it a bit more obvious what is being used, also with the way it is now any other functions exported from `errorController.ts` would implicitly get used here, which might seem like unexpected behaviour/confusing if someone doesn't know about it.  Not too big of a deal though! just a suggestion.

**Review by tferns-ch — DISMISSED** [review]

File: src/app.ts (hunk 1 above)
Comment (mmodzelewski-ch) [reply src/app.ts]: Done as suggested

**Review by tferns-ch — APPROVED** [review]

**Review by kkonuganti-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by bwallace-ch — APPROVED** [review]

**Review by kkonuganti-ch — APPROVED** [review]

**Pull request merged**